File: backend/src/main/java/org/nure/gotrip/model/Participating.java
package org.nure.gotrip.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Getter
@Setter
@Entity
@Table(name = "participating")
public class Participating {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "participating_id")
	private long id;

	@JsonIgnore
	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "registered_user_id")
	private RegisteredUser user;

	@JsonIgnore
	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "tour_id")
	private Tour tour;

	@Column(name = "tour_rate")
	private int tourRate;

	@Column(name = "guide_rate")
	private int guideRate;

	@Column(name="order_id")
	private String orderId;

	@Column(name="participated")
	private boolean participated;

	public Participating() {
		//Constructor for JPA
	}
}

File: GoNTrip/GoNTrip/GoNTrip/Model/Participating.cs
using Newtonsoft.Json;

using GoNTrip.ServerInteraction.ModelFieldAttributes;

namespace GoNTrip.Model
{
    public class Participating : ModelElement
    {
        public long id { get; set; }

        [FinishTour]
        public int tourRate { get; set; }

        [FinishTour]
        public int guideRate { get; set; }

        public string orderId { get; set; }
        public bool participated { get; set; }
        public bool finished { get; set; }

        [JsonConstructor]
        public Participating(long id, int tourRate, int guideRate, string orderId, bool participated, bool finished)
        {
            this.id = id;
            this.tourRate = tourRate;
            this.guideRate = guideRate;
            this.orderId = orderId;
            this.participated = participated;
            this.finished = finished;
        }

        public Participating(int tourRate, int guideRate)
        {
            this.tourRate = tourRate;
            this.guideRate = guideRate;
        }
    }
}
